pkg/service/service.go

```diff
@@ -787,7 +787,11 @@ func (s *Service) upsertServiceIntoLBMaps(svc *svcInfo, onlyLocalBackends bool,
 	// Upsert service entries into BPF maps
 	backends := make(map[string]lb.BackendID, len(svc.backends))
 	for _, b := range svc.backends {
-		backends[b.String()] = b.ID
+		// Skip adding the terminating backend to the service map as it
+		// won't be selected to serve new requests.
```

Comment:
Could you also extend the comment a bit in that the backend will still be present in the backends map etc, and the refcounting will take care of the final cleanup once graceful period is over?